pkg/ioutils/readers_test.go

```diff
@@ -2,31 +2,32 @@ package ioutils // import "github.com/docker/docker/pkg/ioutils"
 
 import (
 	"context"
-	"fmt"
+	"errors"
 	"io"
 	"strings"
 	"testing"
 	"time"
-
-	"gotest.tools/v3/assert"
-	is "gotest.tools/v3/assert/cmp"
 )
 
 // Implement io.Reader
 type errorReader struct{}
 
+var expectedErr = errors.New("error reader always fail")
+
 func (r *errorReader) Read(p []byte) (int, error) {
-	return 0, fmt.Errorf("error reader always fail")
+	return 0, expectedErr
 }
```

Comment:
This could all be replaced with an [`iotest.ErrReader`](https://pkg.go.dev/testing/iotest@go1.20.6#ErrReader) instantiated in `TestReaderErrWrapperReadOnError`